tests/deepsparse/transformers/pipelines/test_text_generation.py

```diff
@@ -12,33 +12,238 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+import numpy as np
+import onnx
+import onnxruntime
 from transformers import AutoModelForCausalLM, AutoTokenizer
 
 import pytest
 from deepsparse import Pipeline
+from deepsparse.transformers.utils.helpers import (
+    create_causal_mask,
+    overwrite_onnx_model_inputs,
+)
+from sparsezoo import Model
 
 
-@pytest.mark.slow
-def test_codegen():
-    model_stub = (
-        "zoo:nlg/text_generation/codegen_mono-350m/pytorch/"
-        "huggingface/bigpython_bigquery_thepile/base-none"
-    )
-    pipeline = Pipeline.create(
-        task="text_generation",
-        model_path=model_stub,
-        max_generated_tokens=16,
-        prompt_processing_sequence_length=1,
-        use_deepsparse_cache=False,
+def _initialize_kv_cache_state(model, length=0):
+    # get one of the cache inputs
+    cache_input = next(
+        input for input in model.graph.input if input.name.startswith("past_key_values")
     )
-    prompt = "def fib():"
-    out = pipeline(sequences=prompt)
-    nm_output = prompt + out.sequences[0]
+    # read the shape of the cache input
+    batch_size = cache_input.type.tensor_type.shape.dim[0].dim_value
+    num_attention_heads = cache_input.type.tensor_type.shape.dim[1].dim_value
+    hidden_dims = cache_input.type.tensor_type.shape.dim[3].dim_value
+
+    # create a kv cache dictionary
+    kv_cache = {
+        input_.name: np.zeros(
+            (batch_size, num_attention_heads, length, hidden_dims), dtype=np.float32
+        )
+        for input_ in model.graph.input
+        if input_.name.startswith("past_key_values")
+    }
+
+    return kv_cache
+
+
+@pytest.mark.parametrize(
+    "use_deepsparse_cache",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "model_stub, model_name, uses_bos_token",
+    [
+        (
+            "zoo:nlg/text_generation/opt-1.3b/pytorch/huggingface/opt_pretrain/base",
+            "facebook/opt-1.b",
+            True,
+        ),
+        (
+            "zoo:nlg/text_generation/codegen_mono-350m/pytorch/"
+            "huggingface/bigpython_bigquery_thepile/base-none",
+            "salesforce/codegen-350m-mono",
+            False,
+        ),
+    ],
+    scope="class",
+)
+@pytest.mark.skip(
+    reason="Those tests are too heavy to " "run as a normal part of the CI."
+)
+class TestTextGenerationPipeline:
```

Comment:
All these tests are failing for me:
```
tests/deepsparse/transformers/pipelines/test_text_generation.py::TestTextGenerationPipeline::test_freeze[zoo:nlg/text_generation/opt-1.3b/pytorch/huggingface/opt_pretrain/base-facebook/opt-1.b-True-True] failed with error: Test failed with exception
self = <test_text_generation.TestTextGenerationPipeline object at 0x7f47c6664220>
model_stub = 'zoo:nlg/text_generation/opt-1.3b/pytorch/huggingface/opt_pretrain/base'
model_name = 'facebook/opt-1.b', uses_bos_token = True
use_deepsparse_cache = True
    @pytest.fixture
    def setup(self, model_stub, model_name, uses_bos_token, use_deepsparse_cache):
    
        self.max_generated_tokens = 16
>       self.model = Model(model_stub)
tests/deepsparse/transformers/pipelines/test_text_generation.py:78: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.8/site-packages/sparsezoo/model/model.py:86: in __init__
    _setup_args = self.initialize_model_from_stub(stub=self.source)
.venv/lib/python3.8/site-packages/sparsezoo/model/model.py:355: in initialize_model_from_stub
    ) = load_files_from_stub(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
stub = 'nlg/text_generation/opt-1.3b/pytorch/huggingface/opt_pretrain/base'
valid_params = ['checkpoint', 'deployment', 'recipe', 'recipe_type']
    def load_files_from_stub(
        stub: str,
        valid_params: Optional[List[str]] = None,
    ) -> Optional[
        Tuple[List[Dict[str, Any]], str, Dict[str, str], Dict[str, List[ModelResult]], int]
    ]:
        """
        :param stub: the SparseZoo stub path to the model (optionally
            may include string arguments)
        :param valid_params: list of expected parameter names to be encoded in the
            stub. Will raise a warning if any unexpected param names are given. Leave
            as None to not raise any warnings. Default is None
        :param force_token_refresh: True to refresh the auth token, False otherwise
        :return: The tuple of
            - list of file dictionaries
            - model_id (from the server)
            - parsed param dictionary
            - validation results dictionary
            - compressed model size in bytes
        """
        params = None
        if isinstance(stub, str):
            stub, params = parse_zoo_stub(stub=stub, valid_params=valid_params)
        _LOGGER.debug(f"load_files_from_stub: calling  files from {stub}")
    
        arguments = get_model_metadata_from_stub(stub)
    
        api = GraphQLAPI()
    
        models = api.fetch(
            operation_body="models",
            arguments=arguments,
            fields=[
                "model_id",
                "model_onnx_size_compressed_bytes",
                "files",
                "benchmark_results",
                "training_results",
                "repo_name",
                "repo_namespace",
            ],
        )
    
        matching_models = len(models)
        if matching_models == 0:
>           raise ValueError(
                f"No matching models found with stub: {stub}." "Please try another stub"
            )
E           ValueError: No matching models found with stub: nlg/text_generation/opt-1.3b/pytorch/huggingface/opt_pretrain/base.Please try another stub
.venv/lib/python3.8/site-packages/sparsezoo/model/utils.py:154: ValueError
```